omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentController1_10Test.java

```diff
@@ -257,4 +259,35 @@ public void postAttachment_shouldNotUploadFileAboveSizeLimit() throws Exception
 		// Replay
 		SimpleObject response = deserialize(handle(request));
 	}
+	
+	@Test
+	public void getAttachmentBytes_shouldDownloadFile() throws Exception {
+		
+		// Setup
+		String fileCaption = "Test file caption";
+		String uuid;
+		
+		// Upload Test File
+		String fileName = "testFile.dat";
+		Patient patient = Context.getPatientService().getPatient(2);
+		
+		MockMultipartHttpServletRequest uploadRequest = newUploadRequest(getURI());
+		MockMultipartFile file = new MockMultipartFile("file", fileName, "application/octet-stream", randomData);
+		
+		uploadRequest.addFile(file);
+		uploadRequest.addParameter("patient", patient.getUuid());
+		uploadRequest.addParameter("fileCaption", fileCaption);
+		
+		SimpleObject uploadResponse = deserialize(handle(uploadRequest));
+		uuid = uploadResponse.get("uuid");
+		
+		HttpServletRequest downloadRequest = newGetRequest(getURI() + "/" + uuid + "/bytes");
+		
+		// Replay
+		MockHttpServletResponse downloadResponse = handle(downloadRequest);
+		byte[] downloadedFile = downloadResponse.getContentAsByteArray();
+		
+		// Verify
+		Assert.assertArrayEquals(randomData, downloadedFile);
```

Comment:
You also need to assert that the file metadata that is sent as part of the response is as expected: extension, MIME type, content family... etc.

Reply:
ok